src/renderer/index.tsx

```diff
@@ -46,6 +46,43 @@ class Root extends React.PureComponent<Record<string, never>, State> {
         this.registry = await import('mattermost_webapp/registry');
         this.registry?.setModule<History>('utils/browser_history', createHashHistory());
 
+        // Cookie handling
+        const cookies = await window.mattermost.setupCookies;
+        Object.defineProperty(document, 'cookie', {
+            get() {
+                return this.value || '';
+            },
+            set(cookie) {
+                if (!cookie) {
+                    return '';
+                }
+                const cutoff = cookie.indexOf(';');
+                const pair = cookie.substring(0, cutoff >= 0 ? cutoff : cookie.length);
+                const bits = pair.split('=');
+                const cookies = this.value ? this.value.split('; ') : [];
+
+                // look for an existing cookie and remove it if it exists
+                for (let i = 0; i < cookies.length; i++) {
+                    const cookieBits = cookies[i].split('=');
+                    if (cookieBits[0] === bits[0]) {
+                        cookies.splice(i, 1);
+                        break;
+                    }
+                }
+                cookies.push(pair);
+                this.value = cookies.join('; ');
+                return this.value;
+            },
+        });
```

Comment:
So this is needed because when a local file is loaded in Chrome, cookies are ignored. Since the webapp depends on accessing them using this object in certain places, we need to override to act as a dictionary regardless of what Chrome says. I'll try it with a `Proxy` and get back to you.